app/utilities/UserHelper.java

```diff
@@ -0,0 +1,148 @@
+        user.save();
+    }
+
+    /**
+     * Confirms the user after the link in confirmation email is opened.
+     * @param encodedToken encoded authToken of user
+     * @return returns true
+     */
+    public boolean confirmUser(String encodedToken){
+        String token = decodeToken(encodedToken);
+        User user = User.find.where().like("abh_user_token", token).findUnique();
+        if(user == null)
+            return false;
+
+        user.setConfirmed(true);
+        return true;
+    }
+
+    /**
+     * Decodes data using BASE64 decoding
+     * @param token is the string to be decoded
+     * @return decoded string
+     */
+    public String decodeToken(String token){
+        byte[] valueDecoded= Base64.decodeBase64(token.getBytes());
+        return new String(valueDecoded);
+    }
+
+    /**
+     * encode data  using BASE64 hashing
+     * @param token is string to be encoded
+     * @return encoded string
+     */
+    private String encodeToken(String token){
+        byte[]   bytesEncoded = Base64.encodeBase64(token .getBytes());
+        return new String(bytesEncoded );
+    }
+
+    /**
+     * Creates a token object and initializes it.
+     * Authentication authToken is created randomly and expiration day is set to one day ahead.
+     * @return true if successful, false if unsuccessful
+     */
+    public boolean initializeUser(){
+        if(user != null){
+            user.setAuthToken(new Token());
+            user.getAuthToken().generateToken();
+            user.setConfirmed(false);
+            return true;
+        }
+        return false;
+    }
+
+    public boolean validateUser(){
+        if(validateEmail(user.getEmail()) && validateFirstName(user.getFirstName()) && validateLastName(user.getLastName())
+                && validateGender(user.getGender()) && validatePasswords()){
+            return true;
+        }
+        return false;
+    }
+
+    private boolean validateFirstName( String firstName ){
+        return firstName.matches( "[A-Z][a-zA-Z]*" );
+    }
+    private boolean validateLastName( String lastName ) {
+        return lastName.matches( "[a-zA-z]+([ '-][a-zA-Z]+)*" );
+    }
+    private boolean validateEmail(String mail){
+        String EMAIL_PATTERN = "^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@"
+                        + "[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
+        Pattern pattern = Pattern.compile(EMAIL_PATTERN);
+        Matcher matcher = pattern.matcher(mail);
+        return  matcher.matches();
+    }
+    private boolean validateGender(String gender) {
+        if (gender.equals("male") || gender.equals("female") || gender.equals("MALE") || gender.equals("FEMALE")
+                || gender.equals("Male") || gender.equals("Female")) {
+            return true;
+        }
+        return false;
+    }
+    public boolean validatePasswords(){
+        return user.getPassword().equals(user.getPasswordConfirmation());
+    }
```

Comment:
Use built-in constraints for validation (see: https://www.playframework.com/documentation/2.0/api/java/play/data/validation/Constraints.html) or you can build custom constraints using this API (see: http://stackoverflow.com/a/29241145/3419156).